flow/connectors/clickhouse/clickhouse.go

```diff
@@ -51,19 +81,35 @@ func NewClickhouseConnector(
 		return nil, fmt.Errorf("failed to open connection to Clickhouse peer: %w", err)
 	}
 
+	err = ValidateClickhouse(ctx, database)
+	if err != nil {
+		return nil, fmt.Errorf("invalidated Clickhouse peer: %w", err)
+	}
+
 	pgMetadata, err := metadataStore.NewPostgresMetadataStore(ctx)
 	if err != nil {
 		logger.Error("failed to create postgres metadata store", "error", err)
 		return nil, err
 	}
 
-	s3PeerCreds := utils.S3PeerCredentials{
-		AccessKeyID:     config.AccessKeyId,
-		SecretAccessKey: config.SecretAccessKey,
-		Region:          config.Region,
+	var clickhouseS3Creds *utils.ClickhouseS3Credentials
+	deploymentUID := shared.GetDeploymentUID()
+	flowName, _ := ctx.Value(shared.FlowNameKey).(string)
+	bucketPathSuffix := fmt.Sprintf("%s/%s", deploymentUID, flowName)
+	// Get S3 credentials from environment
+	clickhouseS3Creds = utils.GetClickhouseAWSSecrets(bucketPathSuffix)
+	if clickhouseS3Creds.AccessKeyID == "" &&
```

Comment:
We should prefer user provided s3 credentials to env supplied when user provided available